source/mir/random/discrete.d

```diff
@@ -11,6 +11,212 @@ module mir.random.discrete;
 import std.traits : isNumeric;
 
 /**
+Setup a discrete distribution sampler.
+
+Params:
+    probs = probabilities of the individual, discrete values
+
+Returns:
+    A $(LREF Discrete) sampler that can be called to sample from the distribution.
+*/
+Discrete!T discrete(T)(const(T)[] probs)
+{
+    return Discrete!T(probs);
+}
+
+///
+unittest
+{
+    auto probs = [0.1, 0.2, 0.5, 0.2];
```

Comment:
> You may run into floating-point trouble with some compilers with probabilities based on the summation of 0.1 ;-)

Yes even Vose91 anticipated that and thus inserted the cleanup loops at the end..

> It might be safer/saner to use 10, 20, 50, 20 as your proportions.

I don't know whether there is a huge difference, because 
a) if we multiply `50` with a large n we might run into the same troubles
b) this shifts the problem only a bit - e.g. `50.55` 
c) it's a less intuitive user API